### PR TITLE
Add additional informations to node source

### PR DESCRIPTION
**Pull request by mbenguig**

### CODE REVIEW

File: rm/rm-api/src/main/java/org/ow2/proactive/resourcemanager/common/event/RMNodeSourceEvent.java [hunk 1]
```diff
@@ -150,6 +154,10 @@ public String getSourceDescription() {
         return this.nodeSourceDescription;
     }
 
+    public String getAdditionalInformations() {
+        return additionalInformations;
```
Comment (fviale) [inline rm/rm-api/src/main/java/org/ow2/proactive/resourcemanager/common/event/RMNodeSourceEvent.java]: why not having additionalInformations as Map<String, String> and handle this on the client side (e.g. portal)?

File: rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/NodeSource.java [hunk 2]
```diff
@@ -94,6 +91,35 @@
 @ActiveObject
 public class NodeSource implements InitActive, RunActive {
 
+    public enum AdditionalInformation {
+
+        ACTUAL_COST,
+        COST_LIMIT
+    }
+
+    private Map<AdditionalInformation, String> additionalInformations = null;
```
Comment (fviale) [inline rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/NodeSource.java]: additional information should be generic and a Map<String, String>

File: rm/rm-api/src/main/java/org/ow2/proactive/resourcemanager/common/event/RMNodeSourceEvent.java (hunk 1 above)
Comment (mbenguig) [reply rm/rm-api/src/main/java/org/ow2/proactive/resourcemanager/common/event/RMNodeSourceEvent.java]: Ok

File: rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/NodeSource.java (hunk 2 above)
Comment (mbenguig) [reply rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/NodeSource.java]: Ok

File: rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/NodeSource.java [hunk 3]
```diff
@@ -558,6 +571,57 @@ public void setStatus(NodeSourceStatus status) {
...
+
+    public void removeAndPersistAdditionalInformation(String... keys) {
+        for (String key : keys) {
+            this.additionalInformation.remove(key);
+        }
+        persistAdditionalInformation();
+    }
+
+    private void setRmDbManager(RMDBManager dbManager) {
+        this.dbManager = dbManager;
```
Comment (fviale) [inline rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/NodeSource.java]: this setter seems useless to me

File: rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/NodeSource.java [hunk 4]
```diff
@@ -558,6 +571,57 @@ public void setStatus(NodeSourceStatus status) {
...
+            this.descriptor.getAdditionalInformation().put(key, value);
+            this.monitoring.nodeSourceEvent(new RMNodeSourceEvent(RMEventType.NODESOURCE_UPDATED,
+                                                                  this.administrator.getName(),
+                                                                  this.name,
+                                                                  this.getDescription(),
+                                                                  this.additionalInformation,
+                                                                  this.administrator.getName(),
+                                                                  this.getStatus().toString()));
+            // Persist additional information
+            persistAdditionalInformation();
```
Comment (fviale) [inline rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/NodeSource.java]: persist should be called before sending the event

**Comment by fviale** [comment]
two comments, otherwise LGTM

File: rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/NodeSource.java (hunk 3 above)
Comment (mbenguig) [reply rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/NodeSource.java]: Ok

File: rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/NodeSource.java (hunk 4 above)
Comment (mbenguig) [reply rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/NodeSource.java]: Ok

**Pull request merged**